src/Polysemy/Internal/Union.hs

```diff
@@ -73,59 +82,55 @@ instance Functor (Union r mWoven) where
 
 data Weaving e mAfter resultType where
   Weaving
-    :: forall f e rInitial a resultType mAfter. (Functor f)
+    :: forall t e rInitial a resultType mAfter. (MonadTransControl t)
     => {
-      weaveEffect :: e (Sem rInitial) a
+        weaveEffect :: e (Sem rInitial) a
       -- ^ The original effect GADT originally lifted via
       -- 'Polysemy.Internal.send'.
       -- ^ @rInitial@ is the effect row that was in scope when this 'Weaving'
       -- was originally created.
-      , weaveState :: f ()
-      -- ^ A piece of state that other effects' interpreters have already
-      -- woven through this 'Weaving'. @f@ is a 'Functor', so you can always
-      -- 'fmap' into this thing.
-      , weaveDistrib :: forall x. f (Sem rInitial x) -> mAfter (f x)
-      -- ^ Distribute @f@ by transforming @Sem rInitial@ into @mAfter@. This is
-      -- usually of the form @f ('Polysemy.Sem' (Some ': Effects ': r) x) ->
-      --   Sem r (f x)@
-      , weaveResult :: f a -> resultType
-      -- ^ Even though @f a@ is the moral resulting type of 'Weaving', we
-      -- can't expose that fact; such a thing would prevent 'Polysemy.Sem'
-      -- from being a 'Monad'.
-      , weaveInspect :: forall x. f x -> Maybe x
-      -- ^ A function for attempting to see inside an @f@. This is no
-      -- guarantees that such a thing will succeed (for example,
-      -- 'Polysemy.Error.Error' might have 'Polysemy.Error.throw'n.)
+      , weaveTrans :: forall n x. Monad n => (forall y. mAfter y -> n y) -> Sem rInitial x -> t n x
+      , weaveLowering :: forall z x. Monad z => t z x -> z (StT t x)
```

Comment:
This is the lowering function of a `liftWith`, and corresponds to running a `t z` under the initial state of the computation. So this corresponds to the initial state `f ()` under `weave`.